File: backend/app/schemas/alerts.py
from datetime import date, datetime

from pydantic import BaseModel


class AlertOut(BaseModel):
    type: str
    severity: str
    message: str
    internalSku: str
    lotId: int | None = None
    currentQty: int | None = None
    thresholdQty: int | None = None
    dueDate: date | None = None
    bagOpenedAt: datetime | None = None


class AlertsResponse(BaseModel):
    items: list[AlertOut]
    total: int

File: frontend/src/app/api/alerts.ts
import axios from 'axios';
import { useQuery } from '@tanstack/react-query';

export type AlertSeverity = 'critical' | 'warning';
export type AlertType = 'EXPIRING_LOT' | 'MSL_EXPIRED' | 'LOW_STOCK' | string;

export interface AlertItem {
  type: AlertType;
  severity: AlertSeverity | string;
  message: string;
  internalSku: string;
  lotId: number | null;
  currentQty: number | null;
  thresholdQty: number | null;
  dueDate: string | null;
  bagOpenedAt: string | null;
}

export interface AlertsResponse {
  items: AlertItem[];
  total: number;
}

export function useAlerts(days = 30) {
  return useQuery({
    queryKey: ['alerts', days],
    queryFn: async (): Promise<AlertsResponse> => {
      const { data } = await axios.get('/api/v1/alerts', { params: { days } });
      return {
        items: Array.isArray(data?.items) ? data.items : [],
        total: Number(data?.total ?? 0),
      };
    },
  });
}
